### PR TITLE
[SPARK-45807][SQL] Add createOrReplaceView(..) / replaceView(..) to ViewCatalog

### PR DESCRIPTION
**Pull request by nastra**
### What changes were proposed in this pull request?

ViewCatalog API improvements described in [SPIP](https://docs.google.com/document/d/1XOxFtloiMuW24iqJ-zJnDzHl2KMxipTjJoxleJFz66A/edit?usp=sharing) that didn't make it into the codebase as part of #37556

### Why are the changes needed?

Required for DataSourceV2 view support.

### Does this PR introduce _any_ user-facing change?

No
### How was this patch tested?

N/A

### Was this patch authored or co-authored using generative AI tooling?

N/A

### CODE REVIEW

**Review by jzhuge — APPROVED** [review]
Thanks for catching this!

**Comment by nastra** [comment]
@aokolnychyi could you also take a look at this please?

**Comment by aokolnychyi** [comment]
Will review today.

File: sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/ViewCatalog.java [hunk 1]
```diff
@@ -140,6 +140,87 @@ View createView(
...
+          Identifier ident,
+          String sql,
+          String currentCatalog,
+          String[] currentNamespace,
+          StructType schema,
+          String[] queryColumnNames,
+          String[] columnAliases,
+          String[] columnComments,
+          Map<String, String> properties) throws NoSuchNamespaceException {
+    if (!viewExists(ident)) {
```
Comment (aokolnychyi) [inline sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/ViewCatalog.java:209]: Minor: I think it is easier to read conditions that don't have negation. I'd flip the branches.

**Review by aokolnychyi — APPROVED** [review]
Looks good to me.

**Comment by nastra** [comment]
thanks for reviewing @aokolnychyi, I've applied your suggestion

File: sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/ViewCatalog.java [hunk 2]
```diff
@@ -140,6 +140,92 @@ View createView(
...
+  default void replaceView(
+          Identifier ident,
+          String sql,
+          String currentCatalog,
+          String[] currentNamespace,
+          StructType schema,
+          String[] queryColumnNames,
+          String[] columnAliases,
+          String[] columnComments,
+          Map<String, String> properties) throws NoSuchViewException, NoSuchNamespaceException {
```
Comment (cloud-fan) [inline sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/ViewCatalog.java:170]: shall we have a class `ViewInfo` to wrap these fields? It looks verbose to make so many parameters in 3 methods.

File: sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/ViewCatalog.java [hunk 3]
```diff
@@ -140,6 +140,92 @@ View createView(
...
+   * @param currentCatalog the current catalog
+   * @param currentNamespace the current namespace
+   * @param schema the view query output schema
+   * @param queryColumnNames the query column names
+   * @param columnAliases the column aliases
+   * @param columnComments the column comments
+   * @param properties the view properties
+   * @throws NoSuchNamespaceException If the identifier namespace does not exist (optional)
+   */
+  default void createOrReplaceView(
```
Comment (cloud-fan) [inline sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/ViewCatalog.java:202]: 3 APIs might be too much. How about we follow the create/replace table logical plan? We have a `CreateTable` with a bool field `ignoreIfExists`, and a `ReplaceTable` with a bool field `orCreate`.

So we can have just two methods here
```
def createView(... , ignoreIfExists: Boolean)

def replaceView(..., orCreate: Boolean)
```

File: sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/ViewCatalog.java (hunk 2 above)
Comment (jzhuge) [reply sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/ViewCatalog.java:170]: Agree, it is verbose.

File: sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/ViewCatalog.java (hunk 3 above)
Comment (nastra) [reply sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/ViewCatalog.java:202]: thanks for the feedback @cloud-fan. I'll follow up with your suggestions and will ping you for a review

Comment (nastra) [reply sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/ViewCatalog.java:202]: @cloud-fan I have made the suggested improvements in https://github.com/apache/spark/pull/44330 but I'm not sure about having `ignoreIfExists` on the API when creating a view. 
I checked and the `TableCatalog` also doesn't have such functionality and I wonder if it wouldn't be better to have `ignoreIfExists` be part of the physical plan that creates the view (similar to how it's done in https://github.com/apache/spark/pull/44197/files#diff-d491b3b4f1ac890067f57f4c30cbdecf3585ea91dabf0dced59b31a4e1643462R83)

Comment (cloud-fan) [reply sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/ViewCatalog.java:202]: Makes sense. A simpler API is better. The caller side (Spark) can just ignore the error if `ignoreIfExists` is true.

**Pull request closed without merging**